app/utils/aws.py

```diff
@@ -67,6 +73,47 @@ async def head_s3(bucket: str, key: str) -> bool:
     return response.status_code == 200
 
 
+def get_aws_files(
```

Comment:
If this is blocking, and there's no default limit on the number of files, could someone past a prefix with a bazillion files and just completely block the service?

Reply:
Mmm, you're right. It's not the RIGHT solution, but for the moment I will add a default limit of 10. This means if you specify a prefix filled with 10 things other than valid files, there's a chance you will get a misleading error message. But hey, they should clean up their junk! Also will create a ticket to lay out everything wrong with this approach.

Reply:
Follow-up ticket: https://gfw.atlassian.net/browse/GTC-1434

Reply:
Oh I missed where you set it! Sounds good.